### ISSUE TITLE
Why do flex items and containers overflow <html> and <body> ?

### ISSUE THREAD

**Issue by paxperscientiam** [zurb/foundation-sites#9285]
**How can we reproduce this bug?**
1. Point browser to https://foundation.zurb.com/ (or just see codepen)
2. Open the "Inspector" developer tool.
3. Observe that content overlaps <html> and <body>. Interestingly, the ".off-canvas-wrapper" is not. To that point, I just noticed that every page is wrapped in this way.

**What did you expect to happen?**
I expect content not to overflow <body> and <html>

**What happened instead?**
Said elements are overflowed.

**Test case:**
https://codepen.io/paxperscientiam/pen/kkVYBz

EDIT: grammar

**Comment by paxperscientiam** [comment]
OK, so I've been messing around a lot. I think Foundation is missing a column flowing flex container. As well, off-canvas-menu has not been upgraded into a flex form.

Here is a very rough take on the aforementioned stuff.
https://codepen.io/paxperscientiam/pen/PGvvgO

**Comment by Lindsor** [comment]
Not sure what the solution is but if you remove the:

```css
html,
body {
  height: 100%;
}
```

declaration it fixes the issue.

**Comment by Lindsor** [comment]
@brettsmason Yeah changing it to min-height: 100% fixes the issue.

**Comment by paxperscientiam** [comment]
@Lindsor Defaulting to "auto" height can be problematic when images are involved; the height tends to "jump" to accommodate newly loaded images.

@brettsmason Putting aside the topic of offcanvas for the moment (just for the sake of putting together a minimally working example), using a height of 100% on short pages is problematic.

For example, using `min-height:100%` on `<main></main>` -- eg [this](http://codepen.io/paxperscientiam/pen/GNZpJx) -- goes against the "spirit" of flexbox. Instead of the elements taking up only filling the height of the browser window, it flows beyond.

Trying "toggling" `main {min-height:100%;}` with the above in "Full page" mode and you will see the difference.

Basically, [this](https://www.dropbox.com/s/7kpvb8r2n2jk0u3/Screenshot%202016-11-12%2000.29.48.png?dl=0) versus [this](https://www.dropbox.com/s/zkwmop24m79kr8b/Screenshot%202016-11-12%2000.30.48.png?dl=0).

IMHO, this "beam" idea should be built-in -- as opposed to creating a new class for end-user to futz with.

**Comment by Lindsor** [comment]
@paxperscientiam I dont think im understanding the point of height: 100%. simply removing that completely seems to solve the issue and not introduce another bug.

**Comment by brettsmason** [comment]
@lindsor @paxperscientiam I cant comment too much on flexbox as I dont have much experience using it.
However the reason the html/body is set to 100% for off canvas is for a few reasons.

If the user has a shadow on the content (on by default) then you get a shadow appear underneath (see below). This could potentially be fixed by adding shadow to the off canvas panel as inset instead, though the effect wont be quite the same.
![off-canvas-not-100-percent](https://cloud.githubusercontent.com/assets/73229/20248407/7fc5de96-a9db-11e6-9d3e-d8c42854d8f7.png)

The 2nd issue is the overlay is added to the content, so it set within its constraints. That means the bit below the shadow cant be clicked to close the panel.

Again not sure how we can get around that yet, but if this causes a lot of issues with flexbox, I'll see what can be done.

**Comment by paxperscientiam** [comment]
@brettsmason Hey, too tired to say, but perhaps #9394 might be of interest.

**Comment by brettsmason** [comment]
@paxperscientiam So in my latest commit I removed the 100% height declaration as I don't think its needed anymore for off canvas (further testing required though). Will that solve this issue or is a new class still required for your issue based on your PR?

**Comment by brettsmason** [comment]
@paxperscientiam as you know (thanks for your input!) off canvas has been completely reworked for 6.3 and this issue should now be solved.
Release candidate is out today, witht he full release scheduled for the 15th December.

Closing this, but if you still have issues with 6.3 please re-open.

**Comment by paxperscientiam** [comment]
You're most welcome, @brettsmason ! That said, this flex issue was really only tangentially related to OCM, IIRC.

If I understand correctly, @kball indicated in PR #9434 that something to this effect is being worked on.

Gonna play around with 6.3 now!
